### ISSUE TITLE
Dynamic types are not supported in .NET Standard build

### ISSUE THREAD

**Issue by PKYADAV** [davideicardi/DynamicExpresso#111]
Dear Team, 

While using dynamic types in evaluation string give subject errors. Can it possible or I am doing in wrong way?

Data Object:
`
{
    "id": "TEST10001",
    "name": "TEST CUSTOMER",
    "party_type": "PUBLIC",
    "type": "CUSTOMER",
    "addresses": [],
    "createdby": "SC1005",
    "createddate": "2019-08-05T00:00:00",
    "updatedby": "SC1005",
    "updateddate": "2019-08-05T00:00:00",
    "_rid": "RwgzALX24FcFAAAAAAAAAA==",
    "_self": "dbs/RwgzAA==/colls/RwgzALX24Fc=/docs/RwgzALX24FcFAAAAAAAAAA==/",
    "_etag": "\"00000000-0000-0000-4ed9-1978d73901d5\"",
    "_attachments": "attachments/",
    "_ts": 1565372241
}
`
Want to load address dropdown once the selection changed in customer dropdown e.g. customer.addresses.

**Comment by davideicardi** [comment]
Can you provide more info: error message, code snippet, ...

Consider that ExpandoObject (ie. C# dynamic) is not supported in .NET Standard.

**Comment by PKYADAV** [comment]
Hi @davideicardi, Thanks for your response. Please note, I am using c# dynamic but in .Net Core 3.0. Is it possible in it or not?

Dynamic Class:
```
class DynamicClass : DynamicObject, INotifyPropertyChanged
    {
        readonly IDictionary<string, object> data;

        public DynamicClass()
        {
            this.data = new Dictionary<string, object>();
        }

        public DynamicClass(IDictionary<string, object> source)
        {
            this.data = source;
        }

        public override IEnumerable<string> GetDynamicMemberNames()
        {
            return this.data.Keys;
        }

        public override bool TryGetMember(GetMemberBinder binder, out object result)
        {
            result = this[binder.Name];

            return true;
        }

        public override bool TrySetMember(SetMemberBinder binder, object value)
        {
            this[binder.Name] = value;

            return true;
        }

        public object this[string columnName]
        {
            get
            {
                if (this.data.ContainsKey(columnName))
                {
                    return this.data[columnName];
                }

                return null;
            }
            set
            {
                if (!this.data.ContainsKey(columnName))
                {
                    this.data.Add(columnName, value);

                    this.OnPropertyChanged(columnName);
                }
                else
                {
                    if (this.data[columnName] != value)
                    {
                        this.data[columnName] = value;

                        this.OnPropertyChanged(columnName);
                    }
                }
            }
        }

        private void OnPropertyChanged(string propertyName)
        {
            if (this.PropertyChanged != null)
            {
                this.PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
            }
        }

        public event PropertyChangedEventHandler PropertyChanged;
    }
```
Binding the initial JSON object given above in dynamic object and trying to evaluate. Please advice !!

**Comment by davideicardi** [comment]
No, for now dynamics are supported only on .NET Framework. If I remember well the problem was that on .NET Standard there isn't the `Microsoft.CSharp` assembly that is used to perform create the expression. Maybe now there is some other solution but I'm not investigated it.

**Comment by Fantomik** [comment]
Just need include last Microsoft.CSharp from Nuget for support .NET Standard.

Before:
![image](https://user-images.githubusercontent.com/14327877/72852181-dd271400-3cbe-11ea-8780-146e7eb38237.png)

After:
![image](https://user-images.githubusercontent.com/14327877/72852133-c08adc00-3cbe-11ea-882b-1c05c8f9234e.png)
![image](https://user-images.githubusercontent.com/14327877/72852263-11023980-3cbf-11ea-9363-33e80af850c6.png)

**Comment by davideicardi** [comment]
@Fantomik 
But you are able to compile this also for .NET Standard? In the past `Microsoft.CSharp` was not available for .NET Standard. This lib was present only for .NET Framework I think...
But if now it is available I think that we can use it as in .NET Framework.
Can you try to remove `#if` here and add that dependency (see DynamicExpresso.Core.csproj, remove `condition`).

If you are able to do a PR I will merge it. Thanks!

**Comment by DeluxeAgency2020** [comment]
> @Fantomik
> But you are able to compile this also for .NET Standard? In the past `Microsoft.CSharp` was not available for .NET Standard. This lib was present only for .NET Framework I think...
> But if now it is available I think that we can use it as in .NET Framework.
> Can you try to remove `#if` here and add that dependency (see DynamicExpresso.Core.csproj, remove `condition`).
> 
> If you are able to do a PR I will merge it. Thanks!

@davideicardi I'm able to do that. Can I create PR for my changes?

**Comment by davideicardi** [comment]
> @davideicardi I'm able to do that. Can I create PR for my changes?

Yes sure.

**Comment by malotho** [comment]
Great PR enhancement! This was required from net.adamec.lib.common.dmn.engine.

**Comment by dHamburgler** [comment]
Hi @davideicardi , was there a PR for this? I modified the project locally and it works fine with dynamics. This feature would be great in the latest release.

**Comment by davideicardi** [comment]
No, I didn't receive any PR. But appreciate it if someone was able to enable dynamics also for .NET Standard.

**Comment by dHamburgler** [comment]
PR done! 👍  Supporting JObjects would be an awesome feature too, if someone out there has the time for it.

**Comment by davideicardi** [comment]
Thanks @dHamburgler ! New release 2.3.3 just published!

**Comment by davideicardi** [comment]
Closed by #125
